checkov/terraform/checks/resource/tencentcloud/CVMAllocatePublicIp.py

```diff
@@ -0,0 +1,20 @@
+from checkov.common.models.enums import CheckCategories
+from checkov.terraform.checks.resource.base_resource_value_check import (
+    BaseResourceCheck, CheckResult)
+
+
+class CVMAllocatePublicIp(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure CVM instance should not allocate public IP"
+        id = "CKV_TC_2"
+        supported_resources = ['tencentcloud_instance']
+        categories = [CheckCategories.ENCRYPTION]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict) -> CheckResult:
+        if conf.get("allocate_public_ip") and conf["allocate_public_ip"][0]:
```

Comment:
Default is `false` according to the docs, so the absence this attribute should pass.

Reply:
Only this attribute presence and equal true is fail, others are pass. If this attribute absence is pass. It’s seem no problem

Reply:
You're right.